engine/src/snapshot_api/memory_snapshot.rs

```diff
@@ -34,13 +36,13 @@ impl SnapshotApi for MemorySnapshot {
     }
 
     #[inline]
-    async fn read_exact(&mut self, buf: &mut [u8]) -> std::io::Result<()> {
-        self.data.read_exact(buf).await.map(drop)
+    async fn read_exact(&mut self, buf: &mut BytesMut) -> std::io::Result<()> {
+        read_buf(&mut self.data, buf).await.map(drop)
```

Comment:
Why use `read_buf` instead `read_exact`

Reply:
the `read_exact` function requires that the `buf` to be initialized. Howerver, `BytesMut` instances are not typically initialized, which would compel us to maintain an index if they were.